raptor/pipeline.md

```diff
@@ -0,0 +1,103 @@
+
+### Pipeline
+  1. Prepare metadata input
+     1. If metadata is given as a public URL in `metadataURL`, set `metadataPath` variable to that
+        1. Typewriter can take URL as metadata input
+     1. If metadata is an Azure DevOps artifact
+        1. Download the artifact to a local path
+        2. Save that local path as `metadataPath`
+  2. Build snippet generation tool
+     1. Checkout `microsoft-graph-explorer-api` repo
+     2. Build snippet generation tool
+     3. Save executable's path as `snippetGeneratorPath`
+  3. Checkout `microsoft-graph-docs` repo
+     1. Use `docsBranch` variable if defined
+     2. Use `master` if not defined
+     3. Save repo path as `docsRepoPath`
+  4. Run apidoctor
+     1. checkout `OneDrive/apidoctor` repo
+     2. run `apidoctor` with the following inputs
+        1. `docsRepoPath`
+        2. `snippetGeneratorPath`
+  5. SDK generation
+     1. checkout `msgraph-sdk-dotnet` repo
+        1. Save path as `dotnetRepoPath`
+     2. If `typewriterReleaseVersion` is `true`
+        1. Download `Typewriter` from Github releases
+     3. Else
+        1. Checkout `MSGraph-SDK-Code-Generator` repo
+        2. Build `Typewriter`
+        3. Run Typewriter Tests
+     4. Run `Typewriter` for `CSharp`
+        1. Consumes `metadataPath`
+     5. Clean old generated files in `dotnetRepoPath`
+     6. Copy new generated files into `dotnetRepoPath`
+     7.  Build `Microsoft.Graph` solution
+     8.  Run `Microsoft.Graph` tests
+     9.  Save `Microsoft.Graph.dll` path as `dllPath`
+  6. Raptor
+     1. checkout `msgraph-sdk-raptor` repo
+     2. Reference `Microsoft.Graph.dll` using `dllPath`
+     3. Using `version` information attempt to compile corresponding snippets from `docsRepoPath`
+     4. Give a thumbs up if all tests pass
+
+### Prerequisites for defining the pipeline above
+1. Decouple apidoctor from generating a PR after snippet generation
+   1. Current logic is already running as part of a weekly pipeline. PR generation logic can be moved out of apidoctor.exe to an explicit pipeline step.
+2. Have snippet generation logic as a class library or as an executable for apidoctor's consumption instead of HTTP calls
+3. Have Raptor
+   1. Either reference the project directly in `msgraph-sdk-dotnet` repo
+   2. Or take `Microsoft.Graph.dll` location as an input
+
+## Performance Considerations
+- Is typewriter build time a consideration?
+  - `apidoctor` analysis and extraction of HTTP snippets is the slowest step in all of the proposed steps above and we don't have an in-house solution to speed that up at the moment. This is the step that precedes the HTTP calls to Graph Explorer API for generating snippets. We can consider saving that state as a reusable input if we continuously try to build from `master` branch. Not sure about what the structure of that runtime state looks like.
+
+## Open Issues
+1. Need to sync with AGS team to see where this pipeline is going to be located.
```

Comment:
most likely close to this one
https://microsoftgraph.visualstudio.com/onboarding/_build?definitionId=49&_a=summary